### DISCUSSION TITLE
Size of repaint areas

### DISCUSSION THREAD

**Discussion by hrdl-github** [xournalpp/xournalpp#4071] [General] [answered]
I've been working on making Xournal++ more usability on the Pine64 PineNote, which features an eink panel. After spending some time on the input system I've started working on the output. The newest DRM driver for the display controller works best (i.e. has lower latency) with non-overlapping damage regions. Investigating why the reported damage reasons are so large (usually ~35x35), I stumbled upon this:
https://github.com/xournalpp/xournalpp/blob/962fd4be4f40061462a75404447c5c469e56153e/src/core/gui/PageView.cpp#L695-L699
Is the total margin of 30 pixels on both directions still required? If so, which operations require this margin? Removing these didn't seem to break anything during my fairly limited tests.

**Comment by bhennion** [comment]
I've been wandering about this as well. I don't think this is really useful. At the very best, it's useful but bad code anyway. Note that there is a similar padding in XojPageView::rerenderRect (but in page coordinates - 20mm):
https://github.com/xournalpp/xournalpp/blob/962fd4be4f40061462a75404447c5c469e56153e/src/core/gui/PageView.cpp#L701-L708

You can test a while without those paddings and see if visual artefacts appear. 

If so, whatever tool you've been using to see such artefacts should be fixed (and not the padding reinstated).

**Comment by hrdl-github** [comment]
Thanks, this is exactly what I wanted to hear. I'll keep testing and report back or create a pull request.

**Comment by hrdl-github** [comment, marked as answer]
Addressed by https://github.com/xournalpp/xournalpp/pull/4493.
